code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -0,0 +1,253 @@
+// ***************************************
+// *********** Drain blood
+// ***************************************
+/datum/action/xeno_action/activable/drain
+	name = "Drain"
+	action_icon_state = "drain"
+	mechanics_text = "Hold a marine and drain some of their blood if successful."
+	cooldown_timer = 15 SECONDS
+	plasma_cost = 0
+	target_flags = XABB_MOB_TARGET
+
+/datum/action/xeno_action/activable/drain/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+	if(!iscarbon(A))
+		return FALSE
+	if(!A.can_sting())
+		if(!silent)
+			to_chat(owner, span_xenowarning("We can't drain this!"))
+		return FALSE
+	var/mob/living/carbon/xenomorph/X = owner
+	if(X.plasma_stored >= X.xeno_caste.plasma_max)
+		if(!silent)
+			to_chat(X, span_xenowarning("No need, we feel sated for now..."))
+		return FALSE
+	if(!X.Adjacent(A))
+		if(!silent)
+			to_chat(X, span_xenowarning("It is outside of our reach! We need to be closer!"))
+		return FALSE
+/datum/action/xeno_action/activable/drain/use_ability(mob/living/carbon/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	X.face_atom(A)
+	X.do_attack_animation(A, ATTACK_EFFECT_GRAB)
+	X.emote("roar")
+	A.drop_all_held_items()
+	X.visible_message(A, span_danger("The [X] grabs [A]!"))
+	A.SetImmobilized(2 SECONDS)
+	if(!do_after(X, 2 SECONDS, TRUE, A, BUSY_ICON_HOSTILE, ignore_turf_checks = FALSE))
+		to_chat(X, span_xenowarning("Our meal is interrupted!"))
+		succeed_activate()
+		add_cooldown()
+		return
+	X.do_attack_animation(A, ATTACK_EFFECT_REDSTAB)
+	A.emote("scream")
+	A.drop_all_held_items()
+	X.visible_message(A, span_danger("The [X] stabs its tail into [A]!"))
+	A.apply_damage(damage = 20, damagetype = BRUTE, def_zone = BODY_ZONE_HEAD, blocked = 0, sharp = TRUE, edge = FALSE, updating_health = TRUE)
+	playsound(A, "alien_claw_flesh", 25, TRUE)
+	X.gain_plasma(X.xeno_caste.drain_plasma_gain)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Rejuvenate/Transfusion
+// ***************************************
+/datum/action/xeno_action/activable/rejuvenate
+	name = "Rejuvenate/Transfusion"
+	action_icon_state = "rejuvenation"
+	mechanics_text = "When used on self, drains blood and restores health over time. When used on another xenomorph, costs blood and restores some of their health. When used on a dead human, you heal gradually."
+	cooldown_timer = 2 SECONDS
+	plasma_cost = 0
+	use_state_flags = XACT_TARGET_SELF
+	target_flags = XABB_MOB_TARGET
+
+/datum/action/xeno_action/activable/rejuvenate/can_use_ability(atom/target, silent = FALSE, override_flags) //it is set up to only return true on specific xeno or human targets
+	. = ..()
+	if(!.)
+		return
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(H.stat != DEAD)
```

Comment:
no synth check, no robo check. Probably should just check for the NO_BLOOD flag

Reply:
There should be none, maybe synth but robo no